guide/src/pagination.md

```diff
@@ -0,0 +1,127 @@
+any other service.
+
+You may want to follow the service [quickstart]. This guide will walk you
+through the steps necessary to enable the service, ensure you have logged in,
+and that your account has the necessary permissions.
+
+## Dependencies
+
+As it is usual with Rust, you must declare the dependency in your
+`Cargo.toml` file. We use:
+
+```toml
+{{#include ../samples/Cargo.toml:secretmanager}}
+```
+
+## Iterating List methods
+
+To help iterate the items in a list method, the API return an implementation of
+the `Paginator` trait. We need to introduce it in scope via a `use` declaration:
+
+```rust,ignore
+{{#include ../samples/src/pagination.rs:paginator-use}}
+```
+
+To iterate the items, we use the provided Paginator `items` function.
+
+```rust,ignore
+{{#include ../samples/src/pagination.rs:paginator-iterate-items}}
+```
+
+In rare cases, pages contain extra information outside of the items that you may
+need access to. Or you may need to checkpoint your progress across processes. In
+these cases, you can do so by iterating over the full pages instead of the
+individual items.
+
+```rust,ignore
+{{#include ../samples/src/pagination.rs:paginator-iterate-pages}}
+```
+
+### Working with futures::Stream
+
+You may want to use these APIs in the larger Rust ecosystem of asynchronous
+streams, such as `tokio::Stream`.  This is readily done, but you must first
+enable the `unstable-streams` feature in the `google_cloud_gax` crate:
+
+```toml
+{{#include ../samples/Cargo.toml:gax-with-streams}}
+```
+
+The name of this feature is intended to convey that we consider these APIs
+unstable, because they are! You should only use them if you are prepared to work
+with deal with any breaks that result from incompatible changes to the
+[`futures::Stream`][future-stub] trait.
+
+The examples will also use the `futures::stream::StreamExt` trait, so we must
+add the crate that defines it.
+
+```toml
+{{#include ../samples/Cargo.toml:futures}}
+```
+
+```rust,ignore
+{{#include ../samples/src/pagination.rs:paginator-stream-items}}
+```
+
+Similarly, use the `into_stream` function to stream the pages.
+
+```rust,ignore
+{{#include ../samples/src/pagination.rs:paginator-stream-pages}}
+```
+
+## Resuming List methods by setting next page token
+
+In some cases such as an interrupted List operation, we can set the next page
+token to start resume paginating from a specific page.
+
+```rust,ignore
+{{#include ../samples/src/pagination.rs:paginator-page-token}}
+```
+
+## Additional paginator technical details
+
+The standard Google API List method follows the pagination guideline defined by
+[AIP-158]. Each call to a List method for a resource returns a "page" of
+resource items (e.g. Secrets) along with a "next-page token" that can be passed
+to the List method to retrieve the next page.
+
+The Google Cloud Client Libraries for Rust provide an adapter to converts the
+list RPCs as defined by [AIP-4233] into a [`futures::Stream`][future-stub] that
```

Comment:
s/to converts/to convert/

and maybe just say "into a stream that can be ..." instead of mentioning `futures` again, which is optional.